src/Microsoft.IdentityModel.Clients.ActiveDirectory/TokenCache.cs

```diff
@@ -418,6 +418,8 @@ internal AdalResultWrapper LoadFromCacheCommon(CacheQueryData cacheQueryData, Re
                         // this is a cross-tenant result. use RT only
                         resultEx.Result.AccessToken = null;
 
+                        resultEx.Result.Authority = cacheQueryData.Authority;
```

Comment:
>resultEx.Result.Authority = cacheQueryData.Authority; [](start = 24, length = 53)

would that be because it is generally saved with the migrated authority instead of what the user has entered? ... in that case it may not be a cross tenant token which was found but rather simply the base part of the authority which doesn't match... perhaps there should be validation that the tenant is the same vs. the authority base url?

Reply:
i don't think this is needed.

---
In reply to: [241267403](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1461#discussion_r241267403) [](ancestors = 241267403)

Reply:
Did you try your test with-out this one?

---
In reply to: [241910097](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1461#discussion_r241910097) [](ancestors = 241910097,241267403)

Reply:
Yes. I wrote up something earlier today, but I think I forgot to hit "publish". The authority is not part of the data contract on the AdalResult, so when the Clone() is created and the auth result is deserialized, the authority is null. so resultEx.Result.Authority is always null and always returned as null in the silent call. I'm not sure if this is a problem though. That's what I need to figure out. and it's not related to the resource issue. I think they are separate.

Reply:
Makes sense.

---
In reply to: [241939559](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1461#discussion_r241939559) [](ancestors = 241939559)